api/src/test/java/org/openmrs/module/fhir2/providers/r3/ObservationFhirResourceProviderTest.java

```diff
@@ -315,4 +323,197 @@ public void deleteObservation_shouldThrowResourceNotFoundExceptionWhenIdRefersTo
 		
 		resourceProvider.deleteObservationResource(new IdType().setValue(WRONG_OBSERVATION_UUID));
 	}
+	
+	@Test
+	public void getLastnObservations_shouldReturnRecentNObservations() {
+		NumberParam max = new NumberParam(2);
+		ReferenceAndListParam referenceParam = new ReferenceAndListParam();
+		ReferenceParam patient = new ReferenceParam();
+		
+		patient.setValue(PATIENT_UUID);
+		
+		referenceParam.addValue(new ReferenceOrListParam().add(patient));
+		
+		TokenAndListParam categories = new TokenAndListParam().addAnd(new TokenParam().setValue("laboratory"));
+		
+		TokenAndListParam code = new TokenAndListParam().addAnd(
+		    new TokenParam().setSystem(FhirTestConstants.LOINC_SYSTEM_URL).setValue(LOINC_SYSTOLIC_BP),
+		    new TokenParam().setSystem(FhirTestConstants.CIEL_SYSTEM_URN).setValue(CIEL_DIASTOLIC_BP));
+		
+		when(observationService.getLastnObservations(max, referenceParam, categories, code)).thenReturn(
+		    new org.openmrs.module.fhir2.providers.r4.MockIBundleProvider<>(Collections.singletonList(observation), 10, 1));
```

Comment:
We should use the MockIBundleProvider for R3. Please change this throughout the PR.

Reply:
Yes, I'll do that. I apologise for such mistakes.